### ISSUE TITLE
Exceptions should be easier to parse

### ISSUE THREAD

**Issue by cen1** [SparkPost/java-sparkpost#53]
For example, this gets thrown as SparkPostErrorServerResponseException:
`
SEVERE: Error reading server response: { "errors": [ { "message": "Message generation rejected", "description": "recipient address suppressed due to customer policy", "code": "1902" } ], "results": { "total_rejected_recipients": 0, "total_accepted_recipients": 1, "id": "84509310299791535" } }(Bad Request)
`
..which is not really developer friendly. Basically, when exceptions for responses like 4XX are thrown the whole response body is concat to the exception message. If I want to get the internal error code (1902) to store it somewhere I need to first remove the "Error reading server response" from message, then write a custom POJO to parse the remaining body into it.

It would be way nicer if we could grab the error response body directly from SparkPostException as an object, not needing to parse the text message ourself. Or at least get the clean response text.

Right now the library basically hides the response body from us and I don't see any good way to retreive it other than the way mentioned above.

**Comment by yepher** [comment]
This is a great idea. There is already a custom exception so it would not be hard to parse that out at the transport layer and set in member variables.

I would be glad to review a pull request if you are interested in adding this functionality.

**Comment by yepher** [comment]
Adding an array of parsed error responses from server to be more POJO friendly. Here is an example usage:

**Comment by yepher** [comment]
Added several samples that demonstrates the usage of the new field:
    https://github.com/SparkPost/java-sparkpost/blob/feature/better_exceptions_issue53/apps/sparkpost-samples-app/src/main/java/com/sparkpost/error/samples/SendInvalidFromAddress.java#L86


@cen1 is this what you were thinking?

**Comment by cen1** [comment]
@yepher thanks, will check it out in the following days and give feedback.

**Comment by cen1** [comment]
Sorry for late response. Just checked out the master and this is exactly what I wanted. Thank you.
